packages/inventory-next/src/models/reservation-item.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BeforeCreate,
+  Entity,
+  Filter,
+  OnInit,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+
+import { DALUtils } from "@medusajs/utils"
+import { createPsqlIndexStatementHelper } from "@medusajs/utils"
+import { generateEntityId } from "@medusajs/utils"
+
+const ReservationItemDeletedAtIndex = createPsqlIndexStatementHelper({
+  tableName: "reservation_item",
+  columns: "deleted_at",
+  where: "deleted_at IS NOT NULL",
+})
+const ReservationItemLineItemIdIndex = createPsqlIndexStatementHelper({
+  tableName: "reservation_item",
+  columns: "line_item_id",
+})
+
+const ReservationItemInventoryItemIdIndex = createPsqlIndexStatementHelper({
+  tableName: "reservation_item",
+  columns: "inventory_item_id",
+})
+
+const ReservationItemLocationIdIndex = createPsqlIndexStatementHelper({
+  tableName: "reservation_item",
+  columns: "location_id",
+})
+
+@Entity()
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+export class ReservationItem {
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Property({
+    onCreate: () => new Date(),
+    columnType: "timestamptz",
+    defaultRaw: "now()",
+  })
+  created_at: Date
+
+  @Property({
+    onCreate: () => new Date(),
+    onUpdate: () => new Date(),
+    columnType: "timestamptz",
+    defaultRaw: "now()",
+  })
+  updated_at: Date
+
+  @ReservationItemDeletedAtIndex.MikroORMIndex()
+  @Property({ columnType: "timestamptz", nullable: true })
+  deleted_at: Date | null = null
+
+  @ReservationItemLineItemIdIndex.MikroORMIndex()
+  @Property({ type: "text", nullable: true })
+  line_item_id: string | null = null
+
+  @ReservationItemInventoryItemIdIndex.MikroORMIndex()
+  @Property({ type: "text" })
+  inventory_item_id: string
```

Comment:
Should this and the location_id be a foreign key?

Reply:
we can't foreign key location_id as the locations are stored in a separate module

Reply:
Ah I assumed it's part of the inventory module 😄